### ISSUE TITLE
Add maxWait option

### ISSUE THREAD

**Issue by micah-redwood** [unjs/perfect-debounce#31]
### Describe the feature

Hey, first off thanks for this really, really useful async debouncer. Have yet to find another like it. 

Just wanted to suggest adding a `maxWait` option like lodash. Helpful in the case where we want to make sure that we do something at least once every X seconds even if there's pretty much constant activity. 

Some other nice to haves would be to be able to flush the debouncer or at least know if it's currently pending or not. 

Happy to try my hand at a PR if it'd be helpful. Thanks!

### Additional information

- [X] Would you be willing to help implement this feature?

**Comment by peter-facko** [comment]
@pi0, I don't think this is resolved, because `maxWait` feature was not implemented in #35.
